vllm/worker/hpu_worker.py

```diff
@@ -71,6 +71,11 @@ def __init__(
                 not in ["medusa", "mlp_speculator", "eagle"]) \
                     else {"return_hidden_states": True}
 
+        # use VLLM_CONTIGUOUS_PA=false for speculative decoding
+        # otherwise, result is faulty
+        if speculative_config is not None:
+            os.environ["VLLM_CONTIGUOUS_PA"] = "false"
```

Comment:
I'm afraid this won't work reliably. There are 2 places where we check this flag. One is in hpu_model_runner.py and the second one is in ops.py from vllm-hpu-extension and both of these places check that flag only during initalization. This means that we rely on module loading ordering.

Until we figure out why contiguous pa breaks speculative decoding I'd suggest throwing an error if we detect we wan't to have both of those features enabled.

Reply:
It works in my test after using above way to change the ContinuousPA to false and passed correctness test in PR #524 .

the results are something as below => Tested with Medua
using ContinuousPA
```
Spec Decode - latency_per_token is  0.008353662490844727
Generated Text is : [' more than Tamerci auteuroupe week distant landship upwards McGraw)\\ächsuesAM']
```

without ContinuousPA
```
Spec Decode - latency_per_token is  0.008330810070037841
Generated Text is : [' more deeply embedded in the future of humans as distant relatives than that of the future world. AAM']
```

Reply:
Hi, lets first understand why Spec Decode, which is introduced in this PR, does not work without disabling continuousPA

Reply:
I checked that only places that we construct VLLMKVCache (which reads VLLM_CONTIGUOUS_PA env variable) are HabanAttentionBackend at this moment, which is called in Attention class in vllm/attention/layer.py. Since you call disabling contiguousPA in hpu worker constructor, before load_model method is called (that calls Attn constr) I think we could keep it here.